src/repo/github.go

```diff
@@ -0,0 +1,51 @@
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"bytes"
+	"time"
+	"encoding/json"
+)
+
+type TokenIssuer interface {
+	GeneratedJwtToken() (string, error)
+}
+type GithubClient struct {
+	tokenIssuer TokenIssuer
+}
+func NewGithubClient(tokenIssuer TokenIssuer) *GithubClient {
+	return &GithubClient{
+	  tokenIssuer: tokenIssuer,
+	}
+  }
+
+func (c *GithubClient)  IssueAccessToken(installationID string) (string,error) {
+	jwtToken,err := c.tokenIssuer.GeneratedJwtToken()
+	if(err != nil){
+	return "",fmt.Errorf("%s",err)
+	}
+	url :=fmt.Sprintf("https://api.github.com/app/installations/%s/access_tokens",installationID)
+	req, err := http.NewRequest("POST", url,bytes.NewBuffer(nil))
+	req.Header.Set("Authorization","Bearer"+jwtToken)
+	req.Header.Set("Accept", "application/vnd.github+json")
+	req.Header.Set("X-GitHub-Api-Version", "2022-11-28")
+	
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+			return "", fmt.Errorf("failed to execute request: %w", err)
+	}
+	defer resp.Body.Close()
+	
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("non-200 response: %d", resp.StatusCode)
+	}
+	var responseBody struct{
```

Comment:
I would rather define the data type outside of the function.
and it's good to have explicit json tags like `json:"token"`

Reply:
but the scope is inside the function that is why I had define here

Reply:
it makes it anonymous.
you can leave it as is, it's just not how people do in general.

moreover, it might be a case you need to extend a function and return access token + expiresAt timestamp in order to request it only when it's stale, then you need to return a struct and having it anonymous is not ok anymore.

Reply:
understood and suggested changes done